spring-cloud-aws-samples/spring-cloud-aws-sqs-sample/src/main/java/io/awspring/cloud/sqs/sample/SpringSQSManualAck.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2013-2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.sqs.sample;
+
+import io.awspring.cloud.sqs.listener.SqsMessageListenerContainer;
+import io.awspring.cloud.sqs.listener.acknowledgement.Acknowledgement;
+import io.awspring.cloud.sqs.listener.acknowledgement.AcknowledgementResultCallback;
+import io.awspring.cloud.sqs.listener.acknowledgement.handler.AcknowledgementMode;
+import io.awspring.cloud.sqs.operations.SqsTemplate;
+import io.awspring.cloud.sqs.operations.TemplateAcknowledgementMode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.ApplicationRunner;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.messaging.Message;
+import software.amazon.awssdk.services.sqs.SqsAsyncClient;
+
+import java.util.Collection;
+import java.util.UUID;
+
+@Configuration
+public class SpringSQSManualAck {
+
+	public static final String NEW_USER_QUEUE = "new-user-queue";
+
+	// Change to 'true' if you want to not ack.
+	private static final Boolean ACK = false;
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(SpringSQSManualAck.class);
+
+	@Bean
+	public ApplicationRunner sendMessageToQueue(SqsTemplate sqsTemplate) {
+		LOGGER.info("Sending message");
+		return args -> sqsTemplate.sendAsync(to -> to.queue(NEW_USER_QUEUE)
+			.payload(new User(UUID.randomUUID(), "John"))
+		).thenAccept(result -> LOGGER.info("Message sent"));
+	}
+
+	@Bean
+	public SqsTemplate sqsOperations(SqsAsyncClient sqsAsyncClient) {
+		return SqsTemplate.builder()
+			.sqsAsyncClient(sqsAsyncClient)
+			.configure(options -> options.acknowledgementMode(TemplateAcknowledgementMode.MANUAL))
+			.build();
+	}
+
+	@Bean
+	SqsMessageListenerContainer<User> sqsMessageListenerContainer(SqsAsyncClient sqsAsyncClient) {
+		SqsMessageListenerContainer<User> container = new SqsMessageListenerContainer<>(sqsAsyncClient);
```

Comment:
While this is a perfectly good use case for manual ack, I think it illustrates more instantiating a container manually - which is a more advanced feature - than the manual ack itself.

So I'm thinking we keep this one as a Manual Container Instantiation sample, and maybe create another one configuring the manual ack in the MessageListenerContainerFactory and using the regular `@SqsListener`?

What do you think?

Reply:
Perfect, I already created both here and I can add it in the next PR.